chromium_src/tools/web_dev_style/eslint.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2022 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import os
+
+import brave_node
+import override_utils
+
+
+@override_utils.override_function(globals())
+def Run(_original_function, **kwargs):
+    node_args = [
+        brave_node.PathInNodeModules('eslint', 'bin', 'eslint'),
+        '--quiet',
+        '--resolve-plugins-relative-to',
```

Comment:
I guess not https://github.com/brave/brave-core/pull/16289/files#diff-57765a2557bc107c8e5ef5c7c256163c899314e0e7ad9ec7ba31d9f5188d0bafR7

Reply:
Actually it seems like we will because that eslint only applies to certain directories. Isn't there some way to specify additional plugin directories? Or can we make these relative paths? https://github.com/brave/brave-core/pull/16289/files#diff-57765a2557bc107c8e5ef5c7c256163c899314e0e7ad9ec7ba31d9f5188d0bafR69

Reply:
> does this mean we have to make sure we include any plugins used by upstream in our node modules? What are they currently missing that we need?

yes we need to add missing modules that upstream uses for their `eslint` runs. 

`brave/components/.eslintrc.js` actually should not have `'root': true`, but currently we need this to gradually inherit upstream rules.

this is the list of upstream modules (located in `src/third_party/node/package.json`):
```
{
  "name": "webui-node-modules",
  "version": "1.0.0",
  "author": "dpapad@chromium.org",
  "dependencies": {
    "@types/chai": "4.2.22",
    "@types/d3": "5.16.0",
    "@types/dom-mediacapture-record": "1.0.11",
    "@types/dom-speech-recognition": "0.0.1",
    "@types/filesystem": "0.0.32",
    "@types/filewriter": "0.0.29",
    "@types/google.analytics": "0.0.42",
    "@types/mocha": "9.0.0",
    "@types/offscreencanvas": "2019.6.4",
    "@types/trusted-types": "1.0.6",
    "@types/w3c-css-typed-object-model-level-1": "20180410.0.4",
    "@types/w3c-image-capture": "1.0.4",
    "@types/webrtc": "0.0.31",
    "@typescript-eslint/eslint-plugin": "5.12.0",
    "@typescript-eslint/parser": "5.12.0",
    "babel-eslint": "10.0.2",
    "eslint": "7.11.0",
    "eslint-plugin-jsdoc": "37.5.1",
    "html-minifier": "4.0.0",
    "rollup": "2.58.0",
    "svgo": "2.8.0",
    "terser": "5.3.3",
    "typescript": "4.6.3"
  }
}
```

we only have this (`brave/package.json`): `"@typescript-eslint/eslint-plugin": "4.33.0",`

Reply:
> Actually it seems like we will because that eslint only applies to certain directories. Isn't there some way to specify additional plugin directories? Or can we make these relative paths? 

no, there's no way to use multiple directories for plugins. We just need to update/add deps during Chromium rebase if `eslint` will start fail.